server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -0,0 +1,175 @@
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> PORT = intSetting(REMOTE_CLUSTER_PREFIX + "port", 9900, 0, 65535, Setting.Property.NodeScope);
+
+    public static final Setting<Integer> PUBLISH_PORT = intSetting(
+        REMOTE_CLUSTER_PREFIX + "publish_port",
+        -1,
+        -1,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Boolean> TCP_KEEP_ALIVE = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_alive",
+        TransportSettings.TCP_KEEP_ALIVE,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> TCP_KEEP_IDLE = intSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_idle",
+        TransportSettings.TCP_KEEP_IDLE,
+        -1,
+        300,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> TCP_KEEP_INTERVAL = intSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_interval",
+        TransportSettings.TCP_KEEP_INTERVAL,
+        -1,
+        300,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> TCP_KEEP_COUNT = intSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_count",
+        TransportSettings.TCP_KEEP_COUNT,
+        -1,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Boolean> TCP_NO_DELAY = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.no_delay",
+        TransportSettings.TCP_NO_DELAY,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Boolean> TCP_REUSE_ADDRESS = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.reuse_address",
+        TransportSettings.TCP_REUSE_ADDRESS,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<ByteSizeValue> TCP_SEND_BUFFER_SIZE = Setting.byteSizeSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.send_buffer_size",
+        TransportSettings.TCP_SEND_BUFFER_SIZE,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<ByteSizeValue> TCP_RECEIVE_BUFFER_SIZE = Setting.byteSizeSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.receive_buffer_size",
+        TransportSettings.TCP_RECEIVE_BUFFER_SIZE,
+        Setting.Property.NodeScope
+    );
+
+    static void validateRemoteAccessSettings(Settings settings) {
+        if (REMOTE_CLUSTER_PORT_ENABLED.get(settings)
+            && settings.getGroups("transport.profiles.", true).keySet().contains(REMOTE_CLUSTER_PROFILE)) {
+            throw new IllegalArgumentException(
+                "Remote Access settings should not be configured using the ["
+                    + REMOTE_CLUSTER_PROFILE
+                    + "] profile. "
+                    + "Use the ["
+                    + REMOTE_CLUSTER_PREFIX
+                    + "] settings instead."
+            );
+        }
+    }
+
+    public static TcpTransport.ProfileSettings buildRemoteAccessProfileSettings(Settings settings) {
+        validateRemoteAccessSettings(settings);
```

Comment:
Yes, tests ensuring that things block startup is difficult to do with our test setup, so I just went with the unit test here, but I can confirm that this method is called during node startup, as that's when we configure all the network stuff. I did realize that I could adjust `testRemoteClusterProfileCannotBeUsedWhenRcs2IsEnabled` to call `buildRemoteAccessProfileSettings` (the public method) instead of the internal validation method, and that's a bit more thorough.

Reply:
I pushed a commit (https://github.com/elastic/elasticsearch/pull/91667/commits/e7ade5702ed485e78ebc83f8168a87e33d41c6f2) to add a node startup failure test. 
Also adjusted `testRemoteClusterProfileCannotBeUsedWhenRcs2IsEnabled` to use `buildRemoteAccessProfileSettings` as suggested.